addons/account/tests/common.py

```diff
@@ -194,32 +171,84 @@ def change_company_country(cls, company, country):
             )
 
     @classmethod
-    def setup_company_data(cls, company_name, chart_template=None, **kwargs):
-        ''' Create a new company having the name passed as parameter.
-        A chart of accounts will be installed to this company: the same as the current company one.
-        The current user will get access to this company.
-
-        :param chart_template: The chart template to be used on this new company.
-        :param company_name: The name of the company.
-        :return: A dictionary will be returned containing all relevant accounting data for testing.
-        '''
+    def setup_other_company(cls, **kwargs):
+        # OVERRIDE
+        company = cls._create_company(**{'name': 'company_2'} | kwargs)
+        return cls.collect_company_accounting_data(company)
 
-        company = cls.env['res.company'].create({
-            'name': company_name,
-            **kwargs,
-        })
-        cls.env.user.company_ids |= company
+    @classmethod
+    def setup_independent_company(cls, **kwargs):
+        # OVERRIDE
+        company = super().setup_independent_company(**kwargs)
+        if not company:
+            company = cls._create_company(name='company_1_data', **kwargs)
+        return company
+
+    @classmethod
+    def _create_company(cls, **create_values):
+        country_code = getattr(cls, 'country_code', None)
+        if country_code:
+            country = cls.env['res.country'].search([('code', '=', country_code.upper())])
+            if not country:
+                raise ValueError('Invalid country code')
+            create_values.update({
+                'country_id': create_values.get('country_id', country.id),
+                'currency_id': create_values.get('currency_id', country.currency_id.id),
+            })
```

Comment:
```suggestion
            if 'country_id' not in create_values:
                create_values['country_id'] = country.id
            if 'currency_id' not in create_values and country.currency_id:
                create_values['currency_id'] = country.currency_id.id
```
(cosmetic suggestion)